pandas/core/dtypes/cast.py

```diff
@@ -1494,21 +1494,32 @@ def cast_scalar_to_array(shape, value, dtype: Optional[DtypeObj] = None) -> np.n
 
     Parameters
     ----------
-    shape : tuple
+    shape : tuple or int
     value : scalar value
     dtype : np.dtype, optional
         dtype to coerce
 
     Returns
     -------
-    ndarray of shape, filled with value, of specified / inferred dtype
+    ndarray of shape of list of length shape[1] of Extension Arrays of length shape[0],
```

Comment:
really? why can't we just have an int and the caller can pass shape[1] if needed. this is *really* *really* confusing.

Reply:
To be clear, this is not something that I am changing via this PR, this function previously accepted a `shape` argument, not a `length` argument.

I can't be sure of the motivation here, but I'd guess that this is related to how pandas manages the data in memory. If you have 5 int columns in a DataFrame, you don't want those to be in 5 distinct series since that creates a lot of overhead. Instead you create a single Numpy array of width 5.